kong/plugins/rate-limiting/handler.lua

```diff
@@ -28,10 +28,14 @@ local function get_identifier(conf)
     end
   elseif conf.limit_by == "credential" then
     identifier = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.id
+  elseif conf.limit_by == "http_header" then
+    identifier = ngx.req.get_headers()[conf.http_header]
```

Comment:
We need to handle the case where multiple headers of the same name are sent, and this value is a table (not sure on the correct approach, my initial thought is to return 400, but this needs more thought).